avoid VSINSTALLDIR being prioritized over our own packages

h-vetinari opened the pull request:
Avoid stuff from image getting priority over our own packages; from discussion in https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/408

h-vetinari committed:
avoid VSINSTALLDIR being prioritized over our own packages

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20450235087) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20450235087](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20450235087). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20497551379](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20497551379). Examine the logs at this URL for more detail.</sub>

isuruf commented on recipe/activate.bat:
```diff
@@ -150,6 +150,14 @@ if %ERRORLEVEL% neq 0 (
+IF NOT "%CONDA_BUILD%" == "" (
+  set "PATH=%BUILD_PREFIX%\Library\bin;%PREFIX%\Library\bin;%PATH%"
+) else (
+  set "PATH=%CONDA_PREFIX%\Library\bin;%PATH%"
```
Use all the paths?
```suggestion
  set "PATH=%CONDA_PREFIX%;%CONDA_PREFIX%\Library\bin;%CONDA_PREFIX%\Scripts;%CONDA_PREFIX%\bin;%PATH%"
```

h-vetinari replied:
Never seen `\Library\usr\bin`, but searching revealed that `m2-*` packages are using this. I doubt those would coincide with something in the MSFT images, so IMO they can stay later on the path...? Or can you tell me why `usr/bin` should be first in your opinion? The others additions make sense to me; for now I was mainly checking how I can test this, so this PR wasn't complete yet...

isuruf replied:
I removed `usr/bin`.

h-vetinari committed:
skip test for vs2019

h-vetinari committed:
add all relevant paths for binaries on windows

Co-Authored-By: Isuru Fernando <isuruf@gmail.com>

h-vetinari committed:
log discovered clang version

h-vetinari committed:
avoid python pins

h-vetinari committed:
alphabetize conda-forge.yml

h-vetinari committed:
drop obsolete VS 17.12

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2025.12.22.22.48.19

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10

h-vetinari commented:
17.12 has been superseded for a while now, and it conflicts with newer clang
```
Encountered problems while solving:
  - package clang-21.1.7-default_hac490eb_0 requires vc14_runtime >=14.44.35208, but none of the providers can be installed
```
I suggest we drop it.

h-vetinari committed:
fix order of path extension

h-vetinari committed:
don't break lines without delayed expansion
